org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -337,10 +364,28 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 			if (replace != null && replace.getEnd().getLine() != replace.getStart().getLine()) {
 				replace.setEnd(replace.getStart());
 			}
+			if (itemDefaults.getEditRange() != null && itemDefaults.getEditRange().equals(getEditRange($, preferenceManager))) {
+				$.setTextEditText(newText);
+				$.setTextEdit(null);
+				$.setInsertText(null);
```

Comment:
Have a look at the diff at https://github.com/eclipse/eclipse.jdt.ls/pull/2539#issuecomment-1480199448 . [`insertText`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem) is one of the values no longer kept in the individual completion items. Do we really need to remove it ? It appears to be a fallback used when the `textEdit` lacks both an insert & replace ranges. It might be something other clients rely on.